client/my-sites/themes/install-theme-button.jsx

```diff
@@ -6,79 +6,82 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
-import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
-import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import {
+	isJetpackSiteMultiSite,
+	isJetpackSite,
+	getSiteAdminUrl,
+	getSiteSlug,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
 
 import './install-theme-button.scss';
 
+function getInstallThemeUrl( state, siteId ) {
+	if ( ! siteId ) {
+		return '/themes/upload';
+	}
+
+	const atomicSite = isAtomicSite( state, siteId );
+	const siteCanInstallThemes = siteHasFeature( state, siteId, FEATURE_INSTALL_THEMES );
+	if ( atomicSite && siteCanInstallThemes ) {
```

Comment:
Previously this didn't target only atomic sites. Also, there are other places in the code where users can install themes without the specific atomic check. Can you elaborate on why it is necessary?

Reply:
It did. If you look at the old `connect` function and how it calculates the `siteCanInstallThemes` prop, there is a `siteHasFeature() && atomicSite` condition. I only reproduced it here.